src/HIE/Bios/Cradle.hs

```diff
@@ -235,12 +237,16 @@ getCabalWrapperTool = do
   wrapper_fp <-
     if isWindows
       then do
-        wrapper_hs <- writeSystemTempFile "wrapper.hs" cabalWrapperHs
-        -- the initial contents will be overwritten immediately after by ghc
-        wrapper_fp <- writeSystemTempFile "wrapper.exe" ""
-        let ghc = (proc "ghc" ["-o", wrapper_fp, wrapper_hs])
-                    { cwd = Just (takeDirectory wrapper_hs) }
-        readCreateProcess ghc "" >>= putStr
+        cacheDir <- getXdgDirectory XdgCache "hie-bios"
+        let exeName = ("wrapper-" ++ showVersion version) <.> "exe"
+        let wrapper_fp = cacheDir </> exeName
+        exists <- doesFileExist wrapper_fp
+        unless exists $ do
+            wrapper_hs <- writeSystemTempFile "wrapper.hs" cabalWrapperHs
```

Comment:
Why not make it wrapper-version.hs, and avoid creating differently named files for each which are harder to marry up with their exe?

Reply:
Mmm i am afraid that `writeSystemTempFile` only uses the first 4 letters of the filename (it creates  `wra4EE1.hs` in my system f.e.) so simply adding the version would not work.
We could create directories in `%TEMP` or save the `.hs` in `%XDG_CACHE_HOME` too (and avoid copying it)

Reply:
nvm, we could simply create wrapper-version.hs in TEMP getting ride of `writeSystemTempFile`

Reply:
@ndmitchell i think the pr match your proposal, do you think it is ready to be merged?